test/Test/Toml/BiMap/Property.hs

```diff
@@ -0,0 +1,97 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Toml.BiMap.Property where
+
+import Hedgehog (Gen, PropertyT, Range, assert, forAll, tripping, (===))
+
+import Data.Time (Day, ZonedTime (..))
+import GHC.Natural (Natural)
+import Test.Toml.Gen (PropertyTest, prop)
+import Toml.Bi.Map (BiMap (..), TomlBiMap)
+
+import qualified Data.ByteString.Builder as BB
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.HashSet as HS
+import qualified Data.IntSet as IS
+import qualified Data.List.NonEmpty as NE
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Test.Toml.Gen as G
+import qualified Toml.Bi.Map as B
+
+-- Custom generators
+
+range100 :: Range Int
+range100 = Range.constant 0 100
+
+genNatural :: Gen Natural
+genNatural = Gen.integral (Range.constant 0 (2^(65 :: Int)))
+
+genInSet :: Gen IS.IntSet
+genInSet = IS.fromList <$> Gen.list range100 (Gen.int Range.constantBounded)
+
+genHashSet :: Gen (HS.HashSet Integer)
+genHashSet = HS.fromList <$> Gen.list range100 G.genInteger
+
+genNEDays :: Gen (NE.NonEmpty Day)
+genNEDays = Gen.nonEmpty (Range.constant 1 100) G.genDay
+
+genLByteString :: Gen BL.ByteString
+genLByteString =
+  BB.toLazyByteString . BB.byteString <$>
+  Gen.utf8 range100 Gen.unicodeAll
+
+-- Tests
+
+testBiMap
+    :: (Monad m, Show a, Show b, Eq a)
+    => TomlBiMap a b
+    -> Gen a
+    -> PropertyT m ()
+testBiMap bimap gen = do
+    x <- forAll gen
+    tripping x (forward bimap) (backward bimap =<<)
+
+-- Needs a special test because NaN /= NaN
+test_Double :: PropertyTest
+test_Double =  prop "Double" $ do
+    x   <- forAll G.genDouble
+    if isNaN x
+      then assert $
+          fmap isNaN (forward B._Double x >>= backward B._Double) == Right True
+      else (forward B._Double x >>= backward B._Double) === Right x
+
+test_BiMaps :: PropertyTest
+test_BiMaps = mconcat
```

Comment:
In order to organize all tests into single group you just need to replace this `mconcat` with `testGroup`:

* https://hackage.haskell.org/package/tasty-1.2/docs/Test-Tasty.html#v:testGroup

Like this:

```haskell
test_BiMaps = testGroup "BiMap roundtrip tests"
    [ ...
```

Reply:
Thank you!